worker/uniter/pebblepoller.go

```diff
@@ -95,7 +95,10 @@ func (p *pebblePoller) run(containerName string) error {
 		case <-timer.Chan():
 			timer.Reset(pebblePollInterval)
 			err := p.poll(containerName)
-			if err != nil && err != tomb.ErrDying {
+			socketNotFound := &client.SocketNotFoundError{}
```

Comment:
Nit: might be more idiomatic as `var socketNotFound *client.SocketNotFoundError` because we don't need a value here, `As` will set it (the examples in the [`As` docs](https://pkg.go.dev/errors#As) do it that way too).